crates/miden-tx/src/host/mod.rs

```diff
@@ -530,6 +538,28 @@ impl<'store> TransactionBaseHost<'store> {
         Ok(())
     }
 
+    /// Aborts the transaction by extracting the [`TransactionEffects`] from the stack and returning
+    /// them in an error.
+    ///
+    /// Expected stack state:
+    ///
+    /// ```text
+    /// [ACCOUNT_DELTA_COMMITMENT, INPUT_NOTES_COMMITMENT, OUTPUT_NOTES_COMMITMENT, REPLAY_PROTECTION]
```

Comment:
Now that I think of it more, I wonder if the data on the stack should be something like:
```
[TX_SUMMARY_COMMMITMENT]
```
And then the actual data would be in the advice map:
```
TX_SUMMARY_COMMMITMENT |-> [ACCOUNT_DELTA_COMMITMENT, INPUT_NOTES_COMMITMENT, OUTPUT_NOTES_COMMITMENT, SALT]
```